vault/resource_consul_secret_backend_role.go

```diff
@@ -0,0 +1,170 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func consulSecretBackendRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Create: consulSecretBackendRoleCreate,
+		Read:   consulSecretBackendRoleRead,
+		Update: consulSecretBackendRoleUpdate,
+		Delete: consulSecretBackendRoleDelete,
+		Exists: consulSecretBackendRoleExists,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of an existing role against which to create this Consul credential",
+			},
+			"path": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "Unique name of the Vault Consul mount to configure",
+			},
+			"policies": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "List of Consul policies to associate with this role",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func consulSecretBackendRoleCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	name := d.Get("name").(string)
+	path := d.Get("path").(string)
+	policies := d.Get("policies").([]interface{})
+
+	reqPath := consulSecretBackendRolePath(path, name)
+
+	payload := map[string]interface{}{
+		"policies": policies,
+	}
+
+	d.Partial(true)
+	log.Printf("[DEBUG] Configuring Consul secrets backend role at %q", reqPath)
+
+	d.SetId(path + "," + name)
+
+	if _, err := client.Logical().Write(reqPath, payload); err != nil {
+		return fmt.Errorf("Error writing role configuration for %q: %s", reqPath, err)
```

Comment:
It used to be that in Go, it was preferred that errors start with lowercase. However, I was just checking and I couldn't find the source of that practice, so I'm not sure if it's still the official preference. This repo does largely have them lowercased. 

Not a necessity, but it would be nice to have these start with a lowercase letter. The errors effected are on this line, and 87, 91, 121, 143, and 162.

Reply:
Thanks! I'll be happy to clean these up :+1:

Reply:
Thank you!!!!!